### ISSUE TITLE
no-unused-imports false negative for function when property name matches

### ISSUE THREAD

**Issue by eygraber** [pinterest/ktlint#1915]
I would expect the following code to fail with no-unused-imports:

```kotlin
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.focus.FocusRequester
import androidx.compose.ui.focus.focusRequester

@Composable
fun TestComposable() {
  val focusRequester = remember { FocusRequester() }

  LaunchedEffect(Unit) {
    focusRequester.requestFocus()
  }
}
```

but it does not.

If I rename the property to foo it does fail:

```kotlin
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.focus.FocusRequester
import androidx.compose.ui.focus.focusRequester

@Composable
fun TestComposable() {
  val foo = remember { FocusRequester() }

  LaunchedEffect(Unit) {
    foo.requestFocus()
  }
}
```

I have tested on ktlint 0.47.1 and 0.48.2

**Comment by paul-dingemans** [comment]
In my opinion, this kind of detection is too complicated to be implemented in a linter. There are more problems with detecting unused imports. A more aggressive removal of imports code lead to removal of imports that are actually used. As that would lead to code that does not compile, ktlint has to be conservative when removing imports which are considered to be unused. Also see [this comment](https://github.com/pinterest/ktlint/issues/1754#issuecomment-1368201667).
